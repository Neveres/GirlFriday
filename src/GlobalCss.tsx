import React from 'react'
import { css, Global } from '@emotion/react'

export const GlobalCss = () => (
  <Global
    styles={css`
      * {
        font-family: ${fontFamily.ubuntu};
        font-style: normal;
        color: ${colors.textWhite};
      }

      body {
        margin: 0;
        background-color: ${colors.bgPrimary};
      }

      #root {
        min-height: 100vh;
        position: relative;
      }

      .mobile-page-title {
        font-weight: ${fontWeight.lighter};
        font-size: ${fontSize.xxLarge};
        line-height: 150%;
        margin-bottom: 24px;
      }
    `}
  />
)

export const fontSize = {
  xxxLarge: '48px',
  xxLarge: '24px',
  xLarge: '16px',
  large: '14px',
  medium: '13px',
  small: '12px',
  xSmall: '11px',
}

export const fontWeight = {
  bolder: 700,
  bold: 600,
  lighter: 400,
}

export const fontFamily = {
  ubuntu: 'Ubuntu',
}

export const colors = {
  bgPrimary: '#181818',
  bgSecondary: '#1B1B1B',
  textWhite: '#ffffff',
  textGray: '#b2b2b2',
}
