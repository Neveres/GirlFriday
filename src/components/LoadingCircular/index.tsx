export { default, default as LoadingCircular } from './LoadingCircular'
