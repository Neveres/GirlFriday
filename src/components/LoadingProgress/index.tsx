export { default, default as LoadingProgress } from './LoadingProgress'
