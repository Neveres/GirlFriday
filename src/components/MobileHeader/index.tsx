export { default, default as MobileHeader } from './MobileHeader'
