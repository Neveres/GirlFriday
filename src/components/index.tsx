export { NavBar } from './NavBar'
export { AppBody } from './AppBody'
export { AppContext } from './AppContext'
export { Input } from './Input'
export { Button } from './Button'
export { Slider } from './Slider'
