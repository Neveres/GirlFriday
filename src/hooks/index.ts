export * from './useInnerWidth'
export * from './useTags'
export * from './useUsers'
export * from './useStorage'
export * from './useFriends'
