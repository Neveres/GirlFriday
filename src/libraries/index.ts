export * from './reducer'
export * from './action'
export * from './httpClient'
export * from './storage'
