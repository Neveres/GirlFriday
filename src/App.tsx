import React, { Suspense, useReducer, useMemo } from 'react'
import { useLocation } from 'react-router'
import { NavBar, AppBody, AppContext } from 'src/components'
import { Routes } from 'src/Routes'
import { useDevice } from 'src/hooks'
import { reducer, SET_MOBILE, SET_SEARCH_PARAMETERS } from 'src/libraries'
import { GlobalCss } from './GlobalCss'

const initialState = {
  isMobile: false,
  searchParameters: {
    page: 1,
    pageSize: 10,
    keyword: '',
  } as GirlFriday.SearchParameters,
}

const App = () => {
  const { pathname } = useLocation()

  const [state, dispatch] = useReducer(reducer, initialState)
  const actions = useMemo(
    () => ({
      setSearchParameters(searchParameters: GirlFriday.SearchParameters) {
        dispatch({ type: SET_SEARCH_PARAMETERS, ...state, searchParameters })
      },
      setMobile(isMobile: boolean) {
        dispatch({ type: SET_MOBILE, ...state, isMobile })
      },
    }),
    [state],
  )
  useDevice({ setMobile: actions.setMobile })
  const valueOfAppContext = useMemo(
    () => ({
      state,
      actions,
    }),
    [state, actions],
  )

  return (
    <>
      <AppContext.Provider value={valueOfAppContext}>
        <NavBar pathname={pathname} />

        <AppBody pathname={pathname}>
          <Suspense fallback={<div>Loading...</div>}>
            <Routes />
          </Suspense>
        </AppBody>
      </AppContext.Provider>

      <GlobalCss />
    </>
  )
}

export default App
