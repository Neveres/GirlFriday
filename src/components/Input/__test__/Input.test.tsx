import React from 'react'
import { create } from 'react-test-renderer'
import { Input } from 'src/components'

describe('Input', () => {
  const props = {
    onChange: jest.fn(),
    value: 'value',
    layout: 'desktop',
  }

  test('should render well under desktop layout', () => {
    const testRenderer = create(<Input {...props} />)
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })

  test('should render well under mobile layout', () => {
    props.layout = 'mobile'
    const testRenderer = create(<Input {...props} />)
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
})
