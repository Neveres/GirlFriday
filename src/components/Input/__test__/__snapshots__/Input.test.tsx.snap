// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render well under desktop layout 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9zhpzf-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input css-24rejj-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id=":r0:"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Keyword"
      required={false}
      type="text"
      value="value"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Input should render well under mobile layout 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1tgkr7t-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input css-24rejj-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id=":r1:"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Keyword"
      required={false}
      type="text"
      value="value"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
